### PR TITLE
Make SolidityProxy to dyn resolve compilation result

### PR DESCRIPTION
**Pull request by yann300**
Before that PR, SolidityProxy only has the compilation result of the contract being called and it doesn't know about potential sub-called contract. This update allows SolidityProxy to also decode and interact with contract called during an internal transaction.

the issue arise while fixing debugging proxy contract.

### CODE REVIEW

**Review by bunsenstraat — APPROVED** [review]

File: libs/remix-debug/test/decoder/localDecoder.ts [hunk 1]
```diff
@@ -23,20 +24,53 @@ tape('solidity', function (t) {
 async function test (st, privateKey) {
   var output = compiler.compile(compilerInput(intLocal.contract))
   output = JSON.parse(output)
-  await intLocalTest(st, privateKey, output.contracts['test.sol']['intLocal'].evm.bytecode.object, output, intLocal.contract)
+  let sources = {
+    target: 'test.sol',
+    sources: { 'test.sol': { content: intLocal.contract } }
+  }
+  let compilationResults = new CompilerAbstract('json', output, sources)
+  console.log('intLocalTest')
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/decoder/localDecoder.ts:32]: remove logs

File: libs/remix-debug/test/decoder/localDecoder.ts [hunk 2]
```diff
@@ -23,20 +24,53 @@ tape('solidity', function (t) {
...
+  
   output = compiler.compile(compilerInput(miscLocal.contract))
   output = JSON.parse(output)
-  await miscLocalTest(st, privateKey, output.contracts['test.sol']['miscLocal'].evm.bytecode.object, output)
+  sources = {
+    target: 'test.sol',
+    sources: { 'test.sol': { content: miscLocal.contract } }
+  }
+  compilationResults = new CompilerAbstract('json', output, sources)
+  console.log('miscLocalTest')
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/decoder/localDecoder.ts:42]: same here

File: libs/remix-debug/test/decoder/localsTests/structArray.ts [hunk 3]
```diff
@@ -23,16 +24,28 @@ module.exports = function (st, privateKey, contractBytecode, compilationResult)
...
+          }
+        }
+        var callTree = new InternalCallTree(debuggerEvent, traceManager, solidityProxy, codeManager, { includeLocalVariables: true }, offsetToLineColumnConverter)
         callTree.event.register('callTreeBuildFailed', (error) => {
           st.fail(error)
         })
         callTree.event.register('callTreeReady', (scopes, scopeStarts) => {
           helper.decodeLocals(st, 1622, traceManager, callTree, function (locals) {
             try {
+              console.log('at 1622', locals)
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/decoder/localsTests/structArray.ts:48]: this too

File: libs/remix-debug/test/decoder/localsTests/structArray.ts [hunk 4]
```diff
@@ -98,9 +111,10 @@ module.exports = function (st, privateKey, contractBytecode, compilationResult)
               st.fail(e.message)
             }
           })
-  
+
           helper.decodeLocals(st, 7, traceManager, callTree, function (locals) {
             try {
+              console.log('at 7', locals)
```
Comment (Aniket-Engg) [inline libs/remix-debug/test/decoder/localsTests/structArray.ts:117]: and this

**Review by Aniket-Engg — COMMENTED** [review]
Please remove all the unnecessary logs

File: libs/remix-debug/src/solidity-decoder/solidityProxy.ts [hunk 5]
```diff
@@ -132,8 +123,8 @@ export class SolidityProxy {
    * @param {Int} index  - index of the filename
    * @return {String} - filename
    */
-  fileNameFromIndex (index) {
-    return Object.keys(this.contracts)[index]
+  fileNameFromIndex (index, compilationResult) {
```
Comment (Aniket-Engg) [inline libs/remix-debug/src/solidity-decoder/solidityProxy.ts:126]: devdoc

**Review by Aniket-Engg — APPROVED** [review]

File: libs/remix-debug/src/solidity-decoder/solidityProxy.ts [hunk 6]
```diff
@@ -13,34 +13,21 @@ export class SolidityProxy {
   compilationResult
   sourcesCode
 
-  constructor ({ getCurrentCalledAddressAt, getCode }) {
+  constructor ({ getCurrentCalledAddressAt, getCode, compilationResult }) {
     this.cache = new Cache()
-    this.reset({})
```
[the remark this answers is not in the document]
Comment (yann300) [reply libs/remix-debug/src/solidity-decoder/solidityProxy.ts:18]: not necessary, `reset` only clear the cache now (which is just being instanciated).

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**